src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IObjectCreationExpression.cs

```diff
@@ -1623,7 +2360,75 @@ class C2 : IEnumerable<int>
                         InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
                         OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
 ";
-            VerifyOperationTreeAndDiagnosticsForTest<InitializerExpressionSyntax>(source, expectedOperationTree, expectedDiagnostics, parseOptions: ImplicitObjectCreationOptions);
+            VerifyOperationTreeAndDiagnosticsForTest<InitializerExpressionSyntax>(comp, expectedOperationTree, expectedDiagnostics);
```

Comment:
Aleksey commented:

> It looks like all tests adjusted in the second commit are using target-typed new. I am assuming regular `new` had the same issue. Please confirm.
> It would be good to add verification for a couple of interesting scenarios with a regular `new`.

Reply:
It looks like it doesn't, because the existing code uses a syntax check rather than IOperation check, which I missed on my first pass. I'll update the syntax check instead, we already have coverage of these scenarios for regular object creation.

---
In reply to: [551544303](https://github.com/dotnet/roslyn/pull/50111#discussion_r551544303) [](ancestors = 551544303)